### ISSUE TITLE
v0.4.11 daemon consuming maximum CPU resources

### ISSUE THREAD

**Issue by celeduc** [ipfs/go-ipfs#4294]
#### Version information:
go-ipfs version: 0.4.11-
Repo version: 6
System version: amd64/linux
Golang version: go1.9

#### Description:
In v0.4.10, CPU usage averaged 15% on an Intel® Core™ i7-7500U CPU @ 2.70GHz × 4.  In v0.4.11, CPU usage now averages 350%.

The error log frequently shows "ERROR   floodsub: error reading rpc from <peer.ID XXXXXX>: connection reset comm.go:36" which didn't appear as frequently in v0.4.10.

#### Log output:

> 18:31:24.396 ERROR   floodsub: error reading rpc from <peer.ID RmBhNY>: connection reset comm.go:36
> 18:31:27.211 ERROR   floodsub: error reading rpc from <peer.ID TpVtwL>: connection reset comm.go:36
> 18:31:31.928 ERROR   floodsub: error reading rpc from <peer.ID Wp2Qmp>: connection reset comm.go:36
> 18:31:32.332 ERROR   floodsub: error reading rpc from <peer.ID cYZtwQ>: connection reset comm.go:36
> 18:31:35.475 ERROR   floodsub: already have connection to peer: <peer.ID QJcHxP> floodsub.go:112
> 18:31:43.595 ERROR   floodsub: error reading rpc from <peer.ID bXaSTx>: connection reset comm.go:36
>

**Comment by Stebalien** [comment]
Do you have `--enable-pubsub-experiment` enabled? (Also, we should probably reduce that from an error to an info, the other side resetting the connection isn't really an error per se.

**Comment by celeduc** [comment]
It happens whether `--enable-pubsub-experiment` is set or not.

**Comment by Stebalien** [comment]
Hm. Floodsub shouldn't even be running without that flag being set. I'll look into it.

**Comment by leavehouse** [comment]
I don't know if this is the same issue (so perhaps I should open a separate one) but last night and today I've also had to kill the IPFS daemon because it was consuming max CPU resources.

Both times I saw errors like this:

```
0:56:20.746 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:56:30.763 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port mdns.go:135
10:56:40.711 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port mdns.go:135
10:56:50.724 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:57:00.738 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:57:10.752 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port mdns.go:135
10:57:20.739 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:57:30.715 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port mdns.go:135
10:57:40.711 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:57:50.738 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:58:20.741 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:58:30.713 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:58:40.730 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:58:50.742 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:59:00.751 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:59:30.710 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:59:40.749 ERROR       mdns: mdns lookup error: failed to bind to any unicast udp port mdns.go:135
10:59:50.714 ERROR       mdns: mdns lookup error: failed to bind to any multicast udp port mdns.go:135
```

I do not have ``--enable-pubsub-experiment`` set.

**Comment by Stebalien** [comment]
Actually, could you give me a CPU profile (`curl -o profile 'http://localhost:5001/debug/pprof/profile'`)? I have a feeling that the CPU usage is unrelated (the IPFS network increased 6x over the past week so some interesting pathological behaviors have started to show up).

**Comment by celeduc** [comment]
@Stebalien I've shared the profile via Keybase. I was running with `--enable-pubsub-experiment` set.

**Comment by Stebalien** [comment]
@celeduc sorry, I don't actually use keybase chat (unfortunately, they don't tell you that...) I just use it to distribute my PGP public key. Mind sending me an encrypted email at `steb@protocol.ai`?

**Comment by celeduc** [comment]
Assuming fixed in v0.4.12
